### PR TITLE
Add a mode to LSM where we can limit the size of data in the tree.

### PR DESCRIPTION
**Pull request by agorrod**
A feature request to allow for a high insert throughput into a
table with a size limitation.

Adds a new configuration option to WT_SESSION::create which is
lsm=(chunk_count_limit=0), default to 0 which is disabled.

Refs #1652

### CODE REVIEW

**Comment by agorrod** [comment]
I've tested this lightly with a custom wtperf configuration file:

```
 conn_config="cache_size=500MB"
table_config="lsm=(chunk_size=10MB,chunk_count_limit=5),type=lsm"
icount=5000000
 report_interval=5
 run_time=20
 populate_threads=1
```

File: src/lsm/lsm_meta.c [hunk 1]
```diff
@@ -24,6 +24,9 @@ __wt_lsm_meta_read(WT_SESSION_IMPL *session, WT_LSM_TREE *lsm_tree)
 
 	chunk = NULL;			/* -Wconditional-uninitialized */
 
+	/* LSM trees default to doing merges. */
+	F_SET(lsm_tree, WT_LSM_TREE_MERGES);
```
Comment (michaelcahill) [inline src/lsm/lsm_meta.c]: We could check `F_SET(conn, WT_CONN_LSM_MERGE)` here and fix #1657 as well?

File: src/lsm/lsm_tree.c [hunk 2]
```diff
@@ -378,6 +378,12 @@ __wt_lsm_tree_create(WT_SESSION_IMPL *session,
 	lsm_tree->bloom_bit_count = (uint32_t)cval.val;
 	WT_ERR(__wt_config_gets(session, cfg, "lsm.bloom_hash_count", &cval));
 	lsm_tree->bloom_hash_count = (uint32_t)cval.val;
+	WT_ERR(__wt_config_gets(session, cfg, "lsm.chunk_count_limit", &cval));
+	lsm_tree->chunk_count_limit = (uint64_t)cval.val;
+	if (cval.val == 0)
+		F_SET(lsm_tree, WT_LSM_TREE_MERGES);
```
Comment (michaelcahill) [inline src/lsm/lsm_tree.c]: Do we need to explicitly turn merges on here?

Comment (agorrod) [inline src/lsm/lsm_tree.c]: We don't - since we always re-open after creating a tree, but that is subtle :)

File: src/lsm/lsm_meta.c (hunk 1 above)
Comment (agorrod) [inline src/lsm/lsm_meta.c]: Yes - that sounds right to me.

**Comment by michaelcahill** [comment]
@agorrod, can you please review 9bbb859?

**Comment by agorrod** [comment]
Thanks @michaelcahill The change to bloom filter looks good.

**Pull request merged**